annotation-server/src/common/google-sheets.ts

```diff
@@ -0,0 +1,28 @@
+import { sheets as googleSheets, sheets_v4 } from '@googleapis/sheets';
+
+const sheets = googleSheets('v4');
+
+export async function fetchSpreadsheetCells(
+    spreadsheetId: string,
+    apiKey: string,
+    ranges: string[],
+): Promise<{ value?: string; backgroundColor?: sheets_v4.Schema$Color }[][][]> {
+    const spreadsheetData = await sheets.spreadsheets.get({
+        spreadsheetId: spreadsheetId,
+        key: apiKey,
+        ranges: ranges,
+        includeGridData: true,
+    });
+
+    // this asumes data is in first sheet!
```

Comment:
That's a good question and I just tried it by adding a sheet to the spreadsheet. It doesn't break, because `.sheets` in this case returns only those sheets with matching ranges in the spreadsheet. So the code doesn't assume the data is in the first sheet but that the data is only in one sheet.